### ISSUE TITLE
[BUG] SkiaSharp version 2.88.6 has mosaics when loading images in .NET 7 and the background becomes dark.

### ISSUE THREAD

**Issue by e8-HaopuWang** [mono/SkiaSharp#2650]
### Description

SkiaSharp version 2.88.6 has mosaics when loading images in .NET 7 and the background becomes dark.

![QQ图片20231018151428](https://github.com/mono/SkiaSharp/assets/114386814/ffec1555-8876-4251-ae34-5182477252a6)

Tested SkiaSharp version 2.88.5 without this problem

### Code

The best way to share code for larger projects is a link to a GitHub repository: https://github.com/user/repo/tree/bug-123

But, you can also share a short block of code here:
```cs
// some C# code here
```
You can also share some XAML:
```xaml
<!-- xaml code here -->
```

### Version of SkiaSharp

2.88.3 (Current)

### Last Known Good Version of SkiaSharp

2.88.2 (Previous)

### IDE / Editor

Visual Studio Code (Windows)

### Platform / Operating System

Windows

### Relevant Screenshots

![QQ图片20231018151428](https://github.com/mono/SkiaSharp/assets/114386814/ca53f9ee-23ed-4106-80ff-d8cbf77351f7)
![QQ图片20231018151733](https://github.com/mono/SkiaSharp/assets/114386814/e1d89c24-b0b5-4a0e-959b-f8a0ae11552a)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by NovaKs68** [comment]
It seems to be the same bug as in this ticket #2643

**Comment by IainS1986** [comment]
I might be seeing a similar issues with PDF documents, all DrawBitmap calls are coming out with corrupted images very similar to what you're seeing.

Going to raise a bug tomorrow once I investigate a bit more. Are you using jpgs by any chance? I have my suspicions it's jpg encoding that's broken (and in my case there's no work around as pdf documents convert internal image data to JPGs)

**Comment by IainS1986** [comment]
I could be going down a dead end here but do your images above have alpha channels?

I had a test where I saw some images come out fine, and some corrupt. The corrupt images were all RGB and the fine ones were RGBA.

When I saved one of the fine images to just be RGB, they then also appeared corrupt.

Be good to know if you are using RGB images, and if adding an alpha channel makes any difference.

**Comment by edobnet** [comment]
How do I update?

**Comment by e1em3ntoDX** [comment]
Hi, 

Multiple commercial products at DevExpress also rely on the functionality offered by the SkiaSharp library and we experience similar issues in apps with our tools deployed to macOS/iOS after upgrading to v2.88.6. Downgrading is a no-go for us due to the [vulnerability](https://github.com/mono/SkiaSharp/issues/2608). Is there any ETA of the fix?  

@mattleibow any chance you can look into this and share some details/future plans?

**Comment by jasoncodematic** [comment]
We are using another library which has a dependency on SkiaSharp and this bug is causing lots of issues. Downgrading is also not a great option for us either. Is there anyone currently looking to fix this?

**Comment by AndyG74** [comment]
after downgrade to 2.88.5 it works, but there is a security issue:
![image](https://github.com/mono/SkiaSharp/assets/61390650/6363b0c6-4a22-4e92-94ef-f3303dee2a43)

**Comment by e8-HaopuWang** [comment]
This issue was solved after saving the image in PNG format

**Comment by AndyG74** [comment]
ok, thas fine but i need the image as JPEG
![image](https://github.com/mono/SkiaSharp/assets/61390650/cf6facab-fc29-4a69-8605-043760d615f2)

**Comment by IainS1986** [comment]
@e8-HaopuWang 

This is in no way 'closed'.

**Comment by e8-HaopuWang** [comment]
@AndyG74  I also use JPEG format in versions below 2.88.6, but saving the image as JPEG in version 2.88.6 will be damaged.  After testing, there is no problem in using PNG format.

**Comment by mattleibow** [comment]
I tested 3.0.1 and the latest main and I think this is an issue in libjpeg-turbo. For now I am reverting to the latest 2.x series.

I had a look at the original security issue and it does not affect the 2.x series because 12-bit jpegs are not actually supported in our build. The 2.x series required the entire library to be either 8-bit or 12-bit, and we use 8-bit. The 3.x series allowed for both simultaneously.

The security issue was just reported for 2.1.90 (which seems like a 2.x series but is actually a 3.0 beta). The current stable version is 2.1.5.1 which is on a separate branch.

I have opened an issue in libjpeg-turbo where I will try and determine the real cause and where this issue should live: https://github.com/libjpeg-turbo/libjpeg-turbo/issues/741
